polling_stations/apps/data_importers/base_importers.py

```diff
@@ -0,0 +1,874 @@
+        if self.validation_checks:
+            self.write_context_data()
+        addresses = self.get_addresses()
+        self.write_info(
+            "Addresses: Found {:,} rows in input file".format(len(addresses))
+        )
+        self.write_info("----------------------------------")
+        for address in addresses:
+            address_info = self.address_record_to_dict(address)
+
+            if address_info is None:
+                self.logger.log_message(
+                    logging.INFO,
+                    "address_record_to_dict() returned None with input:\n%s",
+                    variable=address,
+                    pretty=True,
+                )
+                continue
+
+            self.add_residential_address(address_info)
+
+    def add_residential_address(self, address_info):
+
+        if "council" not in address_info:
+            address_info["council"] = self.council
+
+        if "uprn" not in address_info:
+            address_info["uprn"] = ""
+        else:
+            # UPRNs less than 12 characters long may be left padded with zeros
+            # Making sure uprns in our addresslist are not left padded will help with matching them
+            # and catching duplicates.
+            address_info["uprn"] = str(address_info["uprn"]).lstrip("0")
+
+        self.addresses.append(address_info)
+
+
+class BaseStationsDistrictsImporter(BaseStationsImporter, BaseDistrictsImporter):
+    def pre_import(self):
+        raise NotImplementedError
+
+    @property
+    def districts_have_station_ids(self):
+
+        """
+        Check that we've called self.import_polling_{districts,stations}
+        Don't raise an exception though, because we might still want to
+        import just stations or districts for debugging - i.e. to see them
+        in qgis/the db. However if we don't also have the other half we
+        won't be able to update the UprnToCouncil table.
+        """
+        if len(self.districts.elements) < 1:
+            self.logger.log_message(
+                logging.WARNING, "No district records added to self.districts"
+            )
+        if len(self.stations.elements) < 1:
+            self.logger.log_message(
+                logging.WARNING, "No station records added to self.stations"
+            )
+
+        district_ids = {e.internal_council_id for e in self.districts.elements}
+        district_stations_ids = {e.polling_station_id for e in self.districts.elements}
+        station_ids = {e.internal_council_id for e in self.stations.elements}
+        station_districts_ids = {e.polling_district_id for e in self.stations.elements}
+
+        def check_for_mismatches(set_a, set_b):
+            if set_a - set_b:
+                self.logger.log_message(
+                    logging.WARNING,
+                    "No elements in %s have ids for following elements in %s: \n%s",
+                    variable=(set_b, set_a, set_a - set_b),
+                )
```

Comment:
When I was working on this, I found this error message spectacularly unhelpful. I guess you're the one who has to make sense of this when you're working on imports so I'll leave it with you to display the debug info you need (or work it out as you go) but I think there is scope to provide more informative/actionable errors here. `import_bristol` is a good example of a script you can run which does import but has some invalid IDs if you want an example you can play with.